controllers/s3bench/job.go

```diff
@@ -0,0 +1,139 @@
+		checkAndAppendBool(&cmdArgs, opts.Tls, "--tls")
+		checkAndAppendString(&cmdArgs, opts.Region, "--region")
+		checkAndAppendBool(&cmdArgs, opts.Encrypt, "--encrypt")
+		checkAndAppendString(&cmdArgs, opts.Bucket, "--bucket")
+		checkAndAppendString(&cmdArgs, opts.HostSelect, "--host-select")
+		checkAndAppendInt(&cmdArgs, opts.Concurrent, "--concurrent")
+		checkAndAppendBool(&cmdArgs, opts.NoPrefix, "--noprefix")
+		checkAndAppendString(&cmdArgs, opts.BenchOutput, "--benchdata")
+		checkAndAppendString(&cmdArgs, opts.Duration, "--duration")
+		checkAndAppendBool(&cmdArgs, opts.NoClear, "--noclear")
+
+		// TODO: Possibly set this to +1 min from create time or something. May be needed if use clients
+		checkAndAppendString(&cmdArgs, opts.SyncStart, "--syncstart")
+		checkAndAppendBool(&cmdArgs, opts.Requests, "--requests")
+	}
+
+	if (spec.S3ObjectOptions != perfv1alpha1.S3ObjectOptions{}) {
+		opts := spec.S3ObjectOptions
+		checkAndAppendInt(&cmdArgs, opts.Count, "--objects")
+		checkAndAppendString(&cmdArgs, opts.Size, "--obj.size")
+		checkAndAppendString(&cmdArgs, opts.Generator, "--obj.generator")
+		checkAndAppendBool(&cmdArgs, opts.RandomSize, "--obj.randsize")
+
+	}
+
+	if (spec.S3AutoTermOptions != perfv1alpha1.S3AutoTermOptions{}) {
+		opts := spec.S3AutoTermOptions
+
+		if opts.Enabled {
+			checkAndAppendBool(&cmdArgs, opts.Enabled, "--autoterm")
+			checkAndAppendString(&cmdArgs, opts.Duration, "--autoterm.dur")
+			checkAndAppendString(&cmdArgs, opts.Percent, "--autoterm.pct")
+
+		}
+	}
+
+	if (spec.S3AnalysisOptions != perfv1alpha1.S3AnalysisOptions{}) {
+		opts := spec.S3AnalysisOptions
+
+		checkAndAppendString(&cmdArgs, opts.Duration, "--analyze.dur")
+		checkAndAppendString(&cmdArgs, opts.Output, "--analyze.out")
+		checkAndAppendString(&cmdArgs, opts.OperationFilter, "--analyze.op")
+		checkAndAppendBool(&cmdArgs, opts.PrintErrors, "--analyze.errors")
+		checkAndAppendString(&cmdArgs, opts.HostFilter, "--analyze.host")
+		checkAndAppendString(&cmdArgs, opts.Skip, "--analyze.skip")
+		checkAndAppendBool(&cmdArgs, opts.HostDetails, "--analyze.hostdetails")
+	}
+
+	if spec.Mode == "mixed" {
+		if (spec.MixedDistributionOptions != perfv1alpha1.MixedDistributionOptions{}) {
+			checkAndAppendInt(&cmdArgs, spec.MixedDistributionOptions.GetDist, "--get-distrib")
+			checkAndAppendInt(&cmdArgs, spec.MixedDistributionOptions.StatDist, "--stat-distrib")
+			checkAndAppendInt(&cmdArgs, spec.MixedDistributionOptions.PutDist, "--put-distrib")
+			checkAndAppendInt(&cmdArgs, spec.MixedDistributionOptions.DeleteDist, "--delete-distrib")
+		}
+	}
+
+	return cmdArgs
+}
+
+func checkAndAppendInt(cmdArgs *[]string, dist int32, arg string) {
+	if dist > 0 {
+		*cmdArgs = append(*cmdArgs, arg, strconv.Itoa(int(dist)))
+	}
+}
+
+func checkAndAppendBool(cmdArgs *[]string, dist bool, arg string) {
+	if dist {
+		*cmdArgs = append(*cmdArgs, arg)
+	}
+}
+
+func checkAndAppendString(cmdArgs *[]string, dist string, arg string) {
+	if dist != "" {
```

Comment:
It might not matter much for this benchmark, but I think the empty string should be distinguished from a non-specified string. You can do that by using string pointers and checking for the pointer to be nil.